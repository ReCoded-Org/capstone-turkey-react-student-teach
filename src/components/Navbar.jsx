import { Link } from 'react-router-dom';
import { useState } from 'react';
import { GiHamburgerMenu } from 'react-icons/gi';
import { VscChromeClose } from 'react-icons/vsc';

function Navbar() {
  const [dark, setDark] = useState(false);
  const [burger, setBurger] = useState(true);

  return (
    <nav className="flex flex-col justify-center items-center">
      {burger ? (
        <GiHamburgerMenu
          onClick={() => setBurger(!burger)}
          className="burger-icons"
        />
      ) : (
        <VscChromeClose
          onClick={() => setBurger(!burger)}
          className="burger-icons"
        />
      )}
      <div
        className={`lg:flex lg:justify-between lg:items-center w-screen lg:h-[4rem] opacity-${
          burger ? 0 : 100
        } lg:opacity-100 transition-all ease-in-out duration-300`}
      >
        <div>
          <ul className="mt-4 text-2xl lg:mt-0 lg:text-base lg:flex lg:items-center lg:ml-10">
            <li className="mb-3 lg:pr-3 lg:border-r-2 lg:border-cusOrange hover:text-cusOrange transition-all ease-in-out">
              <Link to="/">Home</Link>
            </li>
            <li className="mb-3 lg:px-3 lg:border-r-2 lg:border-cusOrange hover:text-cusOrange transition-all ease-in-out">
              <Link to="/questions">Questions</Link>
            </li>
            <li className="mb-3 lg:px-3 lg:border-r-2 lg:border-cusOrange hover:text-cusOrange transition-all ease-in-out">
              <Link to="/contact">Contact</Link>
            </li>
            <li className="mb-3 lg:px-3 hover:text-cusOrange transition-all ease-in-out">
              <Link to="/about">About</Link>
            </li>
          </ul>
        </div>

        <div className="mt-8 lg:mt-0 hover:scale-110 ease-in-out transition-all">
          <Link
            to="/question+"
            className="cursor-pointer max-w-[10rem] p-3 lg:p-2 rounded-md  bg-cusOrange "
          >
            <span className="text-2xl lg:text-base text-white">
              Ask Question+
            </span>
          </Link>
        </div>

        <div className=" flex flex-col-reverse lg:flex lg:flex-row lg:item-center">
          <div className="flex justify-center mt-5 ml-[3.7rem] lg:mt-0 item-center mr-10">
            <label
              htmlFor="toggleB"
              className="flex item-center cursor-pointer"
            >
              <div className="relative">
                <input
                  type="checkbox"
                  onClick={() => setDark(!dark)}
                  id="toggleB"
                  className="sr-only"
                />

                <div className="bgColor block border-2 border-black-600 w-[4rem] lg:w-[3rem] h-6 rounded-md" />

                <div className="dot absolute left-1 top-1 bg-no-repeat w-9 lg:w-6 h-4 rounded-3xl transition" />
              </div>

              <div className="ml-3 text-red-700 font-medium" />
            </label>
          </div>
          <div>
            <ul className="mt-8 text-2xl  lg:text-base lg:mt-0 flex items-center justify-center ml-10">
              <li>
                <Link
                  to="/signup"
                  className="pr-3 border-r-2 border-cusOrange hover:text-cusOrange transition-all ease-in-out"
                >
                  Sign Up
                </Link>
              </li>
              <li>
                <Link
                  to="/signin"
                  className="cursor-pointer pl-3 mr-10 hover:text-cusOrange transition-all ease-in-out"
                >
                  Sign In
                </Link>
              </li>
            </ul>
          </div>
        </div>
      </div>
    </nav>
  );
}

export default Navbar;
