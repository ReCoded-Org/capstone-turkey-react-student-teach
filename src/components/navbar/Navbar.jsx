import { Link } from 'react-router-dom';
import { useState } from 'react';
import { GiHamburgerMenu } from 'react-icons/gi';
import { VscChromeClose } from 'react-icons/vsc';
import { SiTailwindcss } from 'react-icons/si';
import { FaPlus } from 'react-icons/fa';
import {
  ABOUT_ROUTE,
  CONTACT_ROUTE,
  HOME_ROUTE,
  QUESTIONS_ROUTE,
} from '../../routes';

function Navbar() {
  const [dark, setDark] = useState(false);
  const [burger, setBurger] = useState(true);
  const [addQuestionModal, setAddQuestionModal] = useState(false);
  const [signInModal, setSignInModal] = useState(false);
  const [signUpModal, setSignUpModal] = useState(false);

  return (
    <nav className="flex flex-col justify-center items-center h-[55px] relative">
      <div className="flex flex-row-reverse justify-around items-center w-screen mt-3 lg:mt-0">
        <div>
          {burger ? (
            <GiHamburgerMenu
              onClick={() => setBurger(!burger)}
              className="burger-icons"
            />
          ) : (
            <VscChromeClose
              onClick={() => setBurger(!burger)}
              className="burger-icons"
            />
          )}
        </div>
        <div className="inline-block lg:hidden lg:mr-[6rem] hover:scale-110 ease-in-out transition-all">
          <button
            type="button"
            className="text-sm text-cusOrange lg:text-base p-[7px] rounded-md border-[1px] border-cusOrange lg:p-2 whitespace-nowrap"
          >
            Ask Question
            <FaPlus className="inline-block ml-1" />
          </button>
        </div>
        <SiTailwindcss className="block text-[2rem] lg:hidden " />
      </div>

      <div
        onClick={() => setBurger(true)}
        aria-hidden="true"
        className={`ml-2 w-screen top-[4rem] flex flex-col justify-center items-center lg:ml-0 lg:flex lg:flex-row lg:justify-between lg:items-center lg:h-[4rem] lg:top-[1px] absolute ${
          burger ? 'hidden' : null
        }`}
      >
        <div>
          <ul className="inline-block mt-4 text-2xl lg:mt-0 lg:text-base lg:flex lg:items-center text-center">
            <SiTailwindcss className="hidden mb-3 lg:mb-0 mx-10 text-[2rem] fill-current lg:block" />
            <li className="mb-3 lg:mb-0 lg:pr-3 lg:border-r-[1px] lg:border-cusOrange hover:text-cusOrange transition-all ease-in-out">
              <Link to={HOME_ROUTE}>Home</Link>
            </li>
            <li className="mb-3 lg:mb-0 lg:px-3 lg:border-r-[1px] lg:border-cusOrange hover:text-cusOrange transition-all ease-in-out">
              <Link to={QUESTIONS_ROUTE}>Questions</Link>
            </li>
            <li className="mb-3 lg:mb-0 lg:px-3 lg:border-r-[1px] lg:border-cusOrange hover:text-cusOrange transition-all ease-in-out">
              <Link to={CONTACT_ROUTE}>Contact</Link>
            </li>
            <li className="mb-3 lg:mb-0 lg:px-3 hover:text-cusOrange transition-all ease-in-out">
              <Link to={ABOUT_ROUTE}>About</Link>
            </li>
          </ul>
        </div>

        <div className="hidden mt-5 lg:mt-0 lg:mr-[6rem] lg:inline-block hover:scale-110 ease-in-out transition-all">
          <button
            type="button"
            className="text-sm text-cusOrange lg:text-base p-[7px] rounded-md border-[1px] border-cusOrange lg:p-2 whitespace-nowrap"
          >
            Ask Question
            <FaPlus className="inline-block ml-1" />
          </button>
        </div>

        <div className="flex flex-col-reverse lg:flex lg:flex-row lg:item-center">
          <div className="flex justify-center item-center mt-5 ml-[3rem] mr-10 lg:mt-0">
            <label
              htmlFor="toggleB"
              className="flex item-center cursor-pointer"
            >
              <div className="relative">
                <input
                  type="checkbox"
                  onClick={() => setDark(!dark)}
                  id="toggleB"
                  className="sr-only"
                />

                <div className="bgColor block border-[2px] rounded-md border-cusOrange w-[4rem] h-6 lg:border-[1px] lg:w-[3rem]" />

                <div className="dot left-1 top-1 bg-no-repeat w-9 h-4 rounded-3xl lg:w-6 transition absolute" />
              </div>
            </label>
          </div>
          <div className="inline-block">
            <ul className="flex justify-center items-center mt-2 ml-10 text-2xl lg:text-base lg:mt-0">
              <li>
                <button
                  type="button"
                  className="pr-3 border-r-[2px] border-cusOrange lg:border-r-[1px] hover:text-cusOrange transition-all ease-in-out inline-block"
                >
                  Sign Up
                </button>
              </li>
              <li>
                <button
                  type="button"
                  className="pl-3 mr-10 hover:text-cusOrange transition-all ease-in-out inline-block"
                >
                  Sign In
                </button>
              </li>
            </ul>
          </div>
        </div>
      </div>
    </nav>
  );
}

export default Navbar;
