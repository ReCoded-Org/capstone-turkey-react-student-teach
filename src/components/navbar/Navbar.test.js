import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import Navbar from './Navbar';

it('Navbar snapshot', () => {
  const tree = renderer
    .create(
      <MemoryRouter>
        <Navbar />
      </MemoryRouter>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
