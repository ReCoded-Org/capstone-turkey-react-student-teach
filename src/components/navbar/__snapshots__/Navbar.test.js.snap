// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar snapshot 1`] = `
<nav
  className="flex flex-col justify-center items-center h-[55px] relative"
>
  <div
    className="flex flex-row-reverse justify-around items-center w-screen mt-3 lg:mt-0"
  >
    <div>
      <svg
        className="burger-icons"
        fill="currentColor"
        height="1em"
        onClick={[Function]}
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 512 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M32 96v64h448V96H32zm0 128v64h448v-64H32zm0 128v64h448v-64H32z"
        />
      </svg>
    </div>
    <div
      className="inline-block lg:hidden lg:mr-[6rem] hover:scale-110 ease-in-out transition-all"
    >
      <a
        className="p-[7px] rounded-md border-[1px] border-cusOrange lg:p-2 "
        href="/addQuestion"
        onClick={[Function]}
      >
        <span
          className="text-sm text-cusOrange lg:text-base"
        >
          Ask Question+
        </span>
      </a>
    </div>
    <svg
      className="block text-[2rem] lg:hidden "
      fill="currentColor"
      height="1em"
      role="img"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <title />
      <path
        d="M12.001,4.8c-3.2,0-5.2,1.6-6,4.8c1.2-1.6,2.6-2.2,4.2-1.8c0.913,0.228,1.565,0.89,2.288,1.624 C13.666,10.618,15.027,12,18.001,12c3.2,0,5.2-1.6,6-4.8c-1.2,1.6-2.6,2.2-4.2,1.8c-0.913-0.228-1.565-0.89-2.288-1.624 C16.337,6.182,14.976,4.8,12.001,4.8z M6.001,12c-3.2,0-5.2,1.6-6,4.8c1.2-1.6,2.6-2.2,4.2-1.8c0.913,0.228,1.565,0.89,2.288,1.624 c1.177,1.194,2.538,2.576,5.512,2.576c3.2,0,5.2-1.6,6-4.8c-1.2,1.6-2.6,2.2-4.2,1.8c-0.913-0.228-1.565-0.89-2.288-1.624 C10.337,13.382,8.976,12,6.001,12z"
      />
    </svg>
  </div>
  <div
    aria-hidden="true"
    className="ml-2 w-screen top-[4rem] flex flex-col justify-center items-center lg:ml-0 lg:flex lg:flex-row lg:justify-between lg:items-center lg:h-[4rem] lg:top-[1px] absolute hidden"
    onClick={[Function]}
  >
    <div>
      <ul
        className="inline-block mt-4 text-2xl lg:mt-0 lg:text-base lg:flex lg:items-center text-center"
      >
        <svg
          className="hidden mb-3 lg:mb-0 mx-10 text-[2rem] fill-current lg:block"
          fill="currentColor"
          height="1em"
          role="img"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <title />
          <path
            d="M12.001,4.8c-3.2,0-5.2,1.6-6,4.8c1.2-1.6,2.6-2.2,4.2-1.8c0.913,0.228,1.565,0.89,2.288,1.624 C13.666,10.618,15.027,12,18.001,12c3.2,0,5.2-1.6,6-4.8c-1.2,1.6-2.6,2.2-4.2,1.8c-0.913-0.228-1.565-0.89-2.288-1.624 C16.337,6.182,14.976,4.8,12.001,4.8z M6.001,12c-3.2,0-5.2,1.6-6,4.8c1.2-1.6,2.6-2.2,4.2-1.8c0.913,0.228,1.565,0.89,2.288,1.624 c1.177,1.194,2.538,2.576,5.512,2.576c3.2,0,5.2-1.6,6-4.8c-1.2,1.6-2.6,2.2-4.2,1.8c-0.913-0.228-1.565-0.89-2.288-1.624 C10.337,13.382,8.976,12,6.001,12z"
          />
        </svg>
        <li
          className="mb-3 lg:mb-0 lg:pr-3 lg:border-r-[1px] lg:border-cusOrange hover:text-cusOrange transition-all ease-in-out"
        >
          <a
            href="/"
            onClick={[Function]}
          >
            Home
          </a>
        </li>
        <li
          className="mb-3 lg:mb-0 lg:px-3 lg:border-r-[1px] lg:border-cusOrange hover:text-cusOrange transition-all ease-in-out"
        >
          <a
            href="/questions"
            onClick={[Function]}
          >
            Questions
          </a>
        </li>
        <li
          className="mb-3 lg:mb-0 lg:px-3 lg:border-r-[1px] lg:border-cusOrange hover:text-cusOrange transition-all ease-in-out"
        >
          <a
            href="/contact"
            onClick={[Function]}
          >
            Contact
          </a>
        </li>
        <li
          className="mb-3 lg:mb-0 lg:px-3 hover:text-cusOrange transition-all ease-in-out"
        >
          <a
            href="/about"
            onClick={[Function]}
          >
            About
          </a>
        </li>
      </ul>
    </div>
    <div
      className="hidden mt-5 lg:mt-0 lg:mr-[6rem] lg:inline-block hover:scale-110 ease-in-out transition-all"
    >
      <a
        className="max-w-[10rem] p-3 rounded-md bg-cusOrange lg:p-2"
        href="/addQuestion"
        onClick={[Function]}
      >
        <span
          className="text-2xl text-white lg:text-base"
        >
          Ask Question+
        </span>
      </a>
    </div>
    <div
      className="flex flex-col-reverse lg:flex lg:flex-row lg:item-center"
    >
      <div
        className="flex justify-center item-center mt-5 ml-[3rem] mr-10 lg:mt-0"
      >
        <label
          className="flex item-center cursor-pointer"
          htmlFor="toggleB"
        >
          <div
            className="relative"
          >
            <input
              className="sr-only"
              id="toggleB"
              onClick={[Function]}
              type="checkbox"
            />
            <div
              className="bgColor block border-[2px] rounded-md border-cusOrange w-[4rem] h-6 lg:border-[1px] lg:w-[3rem]"
            />
            <div
              className="dot left-1 top-1 bg-no-repeat w-9 h-4 rounded-3xl lg:w-6 transition absolute"
            />
          </div>
        </label>
      </div>
      <div
        className="inline-block"
      >
        <ul
          className="flex justify-center items-center mt-2 ml-10 text-2xl lg:text-base lg:mt-0"
        >
          <li>
            <a
              className="pr-3 border-r-[2px] border-cusOrange lg:border-r-[1px] hover:text-cusOrange transition-all ease-in-out inline-block"
              href="/signup"
              onClick={[Function]}
            >
              Sign Up
            </a>
          </li>
          <li>
            <a
              className="pl-3 mr-10 hover:text-cusOrange transition-all ease-in-out inline-block"
              href="/signin"
              onClick={[Function]}
            >
              Sign In
            </a>
          </li>
        </ul>
      </div>
    </div>
  </div>
</nav>
`;
